src/lifecycle.js

```diff
@@ -7,7 +7,8 @@
 const MAPPING = {
   'willStart'       : 'getInitialState',
   'willSerialize'   : 'serialize',
-  'willDeserialize' : 'deserialize'
+  'willDeserialize' : 'deserialize',
+  'willReset'       : '__AppWillReset'
```

Comment:
I have decided not to expose this for now. Any ideas on a good lifecycle method name?